lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/fs.rb

```diff
@@ -391,38 +442,9 @@ def cmd_ls(*args)
     columns = [ 'Mode', 'Size', 'Type', 'Last modified', 'Name' ]
     columns.insert(4, 'Short Name') if short
 
-    tbl  = Rex::Ui::Text::Table.new(
-      'Header'  => "Listing: #{path}",
-      'SortIndex' => columns.index(sort),
-      'SortOrder' => order,
-      'Columns' => columns)
-
-    items = 0
-    stat = client.fs.file.stat(path)
+    stat = client.fs.file.stat(/\*|\[|\?/ === path.to_s ? Pathname.new(path).dirname : path)
```

Comment:
it was easier to read before I added the regexp. Also fixed to use the session's version of dirname so we don't get weird behavior running framework on windows.